fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/rule/BaseMaterializedViewRewriteRule.java

```diff
@@ -196,7 +196,7 @@ private PredicateSplit getQuerySplitPredicate(OptimizerContext optimizerContext,
             return null;
         }
         // only add valid predicates into query split predicate
-        Set<ScalarOperator> queryConjuncts = MvUtils.getAllValidPredicates(queryExpression);
+        Set<ScalarOperator> queryConjuncts = MvUtils.getPredicateForRewrite(queryExpression);
```

Comment:
Why not exclude original `isRedundantPredicate` predicates?

Reply:
I filter it by isValidPredicate, already done